avocado/core/messages.py

```diff
@@ -157,6 +159,51 @@ def handle(self, message, task, job):
 class BaseRunningMessageHandler(BaseMessageHandler):
     """Base interface for resolving running messages."""
 
+    _tag = b''
+
+    def __init__(self):
+        self.line_buffer = b''
+
+    def _get_tagged_line(self, data, encoding=None):
+        """
+        It will add the tag in to each line of data as a prefix. If the data
+        don't finish with the new line character, the last line is marked
+        as incomplete, and it is saved to the line buffer for next usage.
+        :param data:
+        :param encoding:
+        :return:
+        """
+        if not encoding:
+            data_lines = data.splitlines(True)
+            if len(data_lines) <= 1 and not data.endswith(b'\n'):
+                self.line_buffer += data
+                return b''
+            else:
+                data = self.line_buffer + b' '.join(data_lines[:-1])
+                if data_lines[-1].endswith(b'\n'):
+                    self.line_buffer = b''
+                    data += data_lines[-1]
+                else:
+                    self.line_buffer = data_lines[-1]
+        return self._tag + self._tag.join(data.splitlines(True))
+
+    def _save_to_default_file(self, message, task):
+        """
+        It will save message log into the default log file.
+
+        The default log file is based on `DEFAULT_LOG_FILE` variable and every
+        line of log will be saved with prefix based on `_tag` variable
+
+        :param message: message from runner
+        :type message: dict
+        :param task: runtime_task which message is related to
+        :type task: :class:`avocado.core.nrunner.Task`
+        """
+        encoding = message.get('encoding', None)
+        data = self._get_tagged_line(message['log'], encoding)
```

Comment:
This method name is indicating that we are going to "get something", but in fact this method is also setting some object property (line 179, 187). I would suggest to refactor this.